cns/restserver/util.go

```diff
@@ -112,9 +113,17 @@ func (service *HTTPRestService) saveNetworkContainerGoalState(req cns.CreateNetw
 	existingNCStatus, ok := service.state.ContainerStatus[req.NetworkContainerid]
 	var hostVersion string
 	var existingSecondaryIPConfigs map[string]cns.SecondaryIPConfig //uuid is key
+	var existingNCVersion string
 	if ok {
 		hostVersion = existingNCStatus.HostVersion
 		existingSecondaryIPConfigs = existingNCStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
+		existingNCVersion = existingNCStatus.VMVersion
+	} else {
+		// Host version is the NC version from NMAgent, set it -2 to indicate no result from NMAgent yet.
+		// Existing NC status will be set as -1 to indicate no version exist yet.
+		// Intentionally set default host version smaller than existing NC version.
+		hostVersion = "-2"
```

Comment:
Are we sure hostVersion represents NC version from NMagent?

Reply:
Why -2 or -1. We must never receive NC GS with Version 0. Set hostVersion to 0

Reply:
1. I'm sure. I checked every place where host version got implemented, set value and referenced. 
2. Can't set it 0. NC version started from 0 in DNC code.